crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -240,55 +298,53 @@ impl DWalletMPCManager {
             .mpc_sessions
             .iter_mut()
             .filter_map(|(_, session)| {
-                let received_weight: PartyID =
-                    if let MPCSessionStatus::Active(round) = session.status {
-                        session.pending_messages[round]
-                            .keys()
-                            .map(|authority_index| {
-                                // Should never be "or"
-                                // as we receive messages only from known authorities.
-                                self.weighted_threshold_access_structure
-                                    .party_to_weight
-                                    .get(authority_index)
-                                    .unwrap_or(&0)
-                            })
-                            .sum()
-                    } else {
-                        0
-                    };
-
-                let is_ready = (matches!(session.status, MPCSessionStatus::Active(_))
-                    && received_weight as StakeUnit >= threshold)
-                    || (session.status == MPCSessionStatus::FirstExecution);
-
-                let mut is_manager_ready = true;
-                if cfg!(feature = "with-network-dkg") {
-                    is_manager_ready = (mpc_network_key_status
-                        == DwalletMPCNetworkKeysStatus::NotInitialized
-                        && matches!(session.party(), MPCParty::NetworkDkg(_)))
+                let received_weight: PartyID = match session.status {
+                    MPCSessionStatus::Active => session
+                        .pending_messages
+                        .get(session.round_number)
+                        .unwrap_or(&HashMap::new())
+                        .keys()
+                        .filter_map(|authority_index| {
+                            self.weighted_threshold_access_structure
+                                .party_to_weight
+                                .get(authority_index)
+                        })
+                        .sum(),
+                    _ => 0,
+                };
+
+                let is_ready = match session.status {
+                    MPCSessionStatus::Active => received_weight as StakeUnit >= threshold,
```

Comment:
In previous pr I said to use the access structure method `is_authorized_subset` for this